src/components/inputs/text/ValidationInput/ValidationInput.tsx

```diff
@@ -0,0 +1,222 @@
+import { useState } from 'react';
+
+import { VALIDATION_INPUT_TYPES } from '../../../../types';
+import {
+  AlertCircle,
+  AlertOctagonIcon,
+  AlertTriangleIcon,
+  CircleCheck,
+} from '../../../icons';
+import { Loader } from '../../../layout';
+import { Tooltip } from '../../../layout/Tooltip/Tooltip';
+import ArweaveTransactionIdInput from '../ArweavetransactionIdInput/ArweavetransactionIdInput';
+
+function ValidationInput({
+  inputType,
+  wrapperClassName,
+  wrapperCustomStyle,
+  showValidationChecklist,
+  inputClassName,
+  inputCustomStyle,
+  placeholder,
+  disabled,
+  value,
+  setValue,
+  setIsValid,
+  setErrors,
+}: {
+  wrapperClassName?: string;
+  wrapperCustomStyle?: any;
+  inputType: VALIDATION_INPUT_TYPES;
+  showValidationChecklist?: boolean;
+  placeholder?: string;
+
+  inputClassName?: string;
+  inputCustomStyle?: any;
+  disabled?: boolean; // disables input
+  value: string;
+  setValue: (text: string) => void;
+  setIsValid: (validity: boolean) => void;
+  setErrors?: (errors: Error[]) => void;
```

Comment:
what about `validationPredicate` as a prop, and that is a function that is passed in but returns errors if validation critiera is not passed.